com.unity.ml-agents.extensions/Runtime/Sensors/ArticulationBodySensor.cs

```diff
@@ -0,0 +1,83 @@
+#if UNITY_2020_1_OR_NEWER
+using UnityEngine;
+using Unity.MLAgents.Sensors;
+
+namespace Unity.MLAgents.Extensions.Sensors
+{
+    /// <summary>
+    /// ISensor implementation to generate observations for a group of ArticulationBodies.
+    /// </summary>
+    public class ArticulationBodySensor : ISensor
+    {
+        int[] m_Shape;
+        string m_SensorName;
+
+        PoseExtractor m_PoseExtractor;
+        PhysicsSensorSettings m_Settings;
+
+        /// <summary>
+        /// Construct a new ArticulationBodySensor.
+        /// </summary>
+        /// <param name="rootBody"></param>
+        /// <param name="settings"></param>
+        /// <param name="sensorName"></param>
+        public ArticulationBodySensor(ArticulationBody rootBody, PhysicsSensorSettings settings, string sensorName=null)
+        {
+            m_PoseExtractor = new ArticulationBodyPoseExtractor(rootBody);
+            m_SensorName = string.IsNullOrEmpty(sensorName) ? $"ArticulationBodySensor:{rootBody?.name}" : sensorName;
```

Comment:
is it typical that rootBody names are unique within the same agent?  I just don't know the workflow.

Reply:
I think it would be rare for there to be more than one root per agent - that would imply e.g. two ragdolls. And then the root would probably be named something like "root" (in Walker, it's "hips").

It's still overridable by the user, this is just a default if they don't provide one.